zeppelin/executors/notebook_executor.py

```diff
@@ -29,10 +29,17 @@ def wait_for_notebook_to_execute(self):
         while True:
             r = requests.get('http://{0}/api/notebook/job/{1}'.format(
                              self.zeppelin_url, self.notebook_id))
+
+            if r.status_code != 200:
+                print('Internal Server Error (500). Trying again in 5 seconds...')
```

Comment:
Also if this could be a 404 or some other status code, maybe a more explicit check of `if r.status_code == 500` instead?